cypress/e2e/spec.cy.js

```diff
@@ -0,0 +1,22 @@
+describe('template spec', () => {
+  beforeEach(() => {
+    cy.visit('/')
+  })
+
+  it('Exibe a saudação "Hi, there" quando nenhum nome é fornecido', () => {
+    cy.get('input[data-testid="name"]').clear()
+    cy.contains('h2','Hi there!').should('be.visible')
+  })
+
+  it('Exibe a saudação "Hi, Joe" quando o nome é fornecido', () => {
+    cy.get('input[data-testid="name"]').clear().type('Joe')
+    cy.contains('h2','Hi Joe!').should('be.visible')
+  })
+
+  it('Abre e fecha o messenger ', () => {
+    cy.get('button[class*=Messenger]').first().should('be.visible').click()
+    cy.contains('How can we help you?').should('be.visible')
+    cy.get('button[class*=Messenger]').last().should('be.visible').click()
+    cy.get('div[class*=Messenger_box]').should('not.exist')
+  })
+})
```

Comment:
Recomenda-se deixar uma linha em branco ao final de cada arquivo (independente da extensão).

Ao deixar tal linha em branco, ferramentas como o GitHub não exibem o ícone de `No newline at end of file`.

A propósito, é possível configurar o ESLint para garantir que tal regra seja sempre cumprida.

```suggestion
})

```